Jolt/Core/ScopeExit.h

```diff
@@ -0,0 +1,49 @@
+// Jolt Physics Library (https://github.com/jrouwe/JoltPhysics)
+// SPDX-FileCopyrightText: 2024 Jorrit Rouwe
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <Jolt/Core/NonCopyable.h>
+
+JPH_NAMESPACE_BEGIN
+
+/// Class that calls a function when it goes out of scope
+template <class F>
+class ScopeExit : public NonCopyable
+{
+public:
+	/// Constructor specifies the exit function
+	JPH_INLINE			ScopeExit(F &&inFunction) : mFunction(inFunction) { }
```

Comment:
I believe you'll want to `std::move(inFunction)` here, to avoid reference collapsing and inadvertently copying the lambda.